modules/slugs/admin/slugs-migrate-qts.php

```diff
@@ -0,0 +1,160 @@
+     *
+     * @param string $table name of the meta table (postmeta, termmeta)
+     * @param string $prefix prefix for the meta key
+     * @param string[] $msg array of messages, updated
+     *
+     * @return void
+     */
+    $count_slugs = function ( $table, $prefix, &$msg ) use ( $wpdb ) {
+        $results = $wpdb->get_var( "SELECT count(*) FROM  $table WHERE meta_key like '$prefix%'" );
+        if ( $results ) {
+            $msg[] = sprintf( __( "Found %d slugs from $table.", 'qtranslate' ), $results );
+        }
+    };
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_META_PREFIX, $msg );
+    if ( ! empty( $msg ) ) {
+        // Found some post/term meta with the new keys, no migrate to suggest (it can still be done manually).
+        return '';
+    }
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+
+    return empty ( $msg ) ? $msg : implode( '<br>', $msg );
+}
+
+/**
+ * Migrate slugs meta by migrating the legacy QTS postmeta and termmeta to QTX.
+ * Attention: current slugs meta are deleted if QTS slugs are found.
+ *
+ * @param bool $db_commit true to commit changes, false for dry-run mode.
+ *
+ * @return string messages giving details.
+ */
+function qtranxf_slugs_migrate_qts_meta( $db_commit ) {
+    global $wpdb;
+
+    $new_prefix = QTX_SLUGS_META_PREFIX;
+    $old_prefix = QTX_SLUGS_LEGACY_QTS_META_PREFIX;
+
+    /**
+     * Generic function that migrates QTS meta to QTX meta.
+     *
+     * @param string $table name of the meta table (postmeta, termmeta)
+     * @param string[] $msg array of messages, updated
+     *
+     * @return void
+     */
+    $migrate_meta = function ( $table, &$msg ) use ( $wpdb, $old_prefix, $new_prefix ) {
+        $results = $wpdb->get_var( "SELECT count(*) FROM  $table WHERE meta_key like '$old_prefix%'" );
+        if ( ! $results ) {
+            $msg[] = sprintf( __( "No slugs to migrate from %s.", 'qtranslate' ), $table );
+        } else {
+            $results = $wpdb->query( "DELETE FROM $table WHERE meta_key like '$new_prefix%'" );
+            $msg[]   = sprintf( __( "Deleted %d slugs from %s (%s).", 'qtranslate' ), $results ?: 0, $table, $new_prefix );
+            // Rename meta keys.
+            $results = $wpdb->query( "UPDATE $table SET meta_key = REPLACE(meta_key, '$old_prefix', '$new_prefix') WHERE meta_key LIKE '$old_prefix%'" );
+            $msg[]   = sprintf( __( "Migrated %d slugs from %s (%s).", 'qtranslate' ), $results ?: 0, $table, $old_prefix );
+        }
+    };
+
+    $msg = [];
+    $wpdb->query( "START TRANSACTION" );
+    $migrate_meta( $wpdb->postmeta, $msg );
+    $migrate_meta( $wpdb->termmeta, $msg );
+    if ( $db_commit ) {
+        $wpdb->query( "COMMIT" );
+    } else {
+        $wpdb->query( "ROLLBACK" );
+    }
```

Comment:
This will work for InnoDB tables only. If table is myISAM changes will be committed even in dry-run mode.
We should assume both scenario are possible.
To cover both we could just count `$new_prefix` and `$old_prefix` to give stats if dry run mode is requested, or fire directly the queries above otherwise (without any explicit START TRANSACTION/COMMIT/ROLLBACK).

Reply:
Good point, I didn't think about that 😕 
Yes it can rewritten differently with pre-counts, now that the migration is simpler. Also, the DELETE can be more narrowed to the post/meta ID that will be renamed.

Reply:
Rewrote the dry-run mode without SQL transaction. 
Add additional filter to narrow the DELETE query by groups of post_id/meta_id to be migrated. Other groups of slugs not concerned by the migration are left unchanged. It should change nothing from a pure migration from scratch but it makes it a bit safer if the admin runs it again later (though it should not).